### DISCUSSION TITLE
Pop! OS 21.04 fails to install when booted via netboot

### DISCUSSION THREAD

**Discussion by Yethal** [ipxe/ipxe#533] [General]
As the title says, when booting Pop! OS 21.04 from netboot the liveCD boots fine but the installation fails right after launch. By comparing output of lsblk to output when ran booted from a usb drive we can see that the iso content did not get mounted under /cdrom. Moreover during early boot we can see that mounting tmpfs on /cdrom fails. please see attached screenshots for more reference

![image](https://user-images.githubusercontent.com/26117918/144643874-bf3b0288-46e7-44dc-ac67-06bca58c276b.png)

![image](https://user-images.githubusercontent.com/26117918/144643389-205841e0-a78d-4fb3-9790-62c10b56f3ed.png)

![image](https://user-images.githubusercontent.com/26117918/144642981-562102f0-8c56-4622-b074-2265770c5159.png)

**Comment by NiKiZe** [comment]
What is in your iPXE script?
And have you read about INT13 as described in https://forum.ipxe.org/showthread.php?tid=5948&pid=19101#pid19101 where memdisk and sanboot is mentioned?

**Reply by Yethal** [reply]
I'm using the stock script that came with netboot.xyz 2.0.53, no custom modifications

**Reply by NiKiZe** [reply]
To help you further I need you to at least post the contents of that script, I will not go and try to find it for you.

**Reply by Yethal** [reply]
This is the template that generates the script:
https://github.com/netbootxyz/netboot.xyz/blob/development/roles/netbootxyz/templates/menu/live-popos.ipxe.j2

This is the generated script:
```
#!ipxe

goto ${menu} ||

:live_menu
set os Pop os Live
menu ${os}
item --gap ${os} Versions
item hirsute ${space} ${os} 21.04
item focal ${space} ${os} 20.04
item bionic ${space} ${os} 18.04
choose live_version || goto live_exit
menu ${os} ${live_version}
item --gap ${os} Flavors
goto ${live_version}

:hirsute
item pop-21.04-default-squash ${space} Pop 21.04 Intel-Amd
item pop-21.04-nvidia-squash ${space} Pop 21.04 Nvidia
goto flavor_select

:focal
item pop-20.04-default-squash ${space} Pop 20.04 Intel-Amd
item pop-20.04-nvidia-squash ${space} Pop 20.04 Nvidia
goto flavor_select

:bionic
item pop-18.04-default-squash ${space} Pop 18.04 Intel-Amd
item pop-18.04-nvidia-squash ${space} Pop 18.04 Nvidia
goto flavor_select

:flavor_select
choose flavor || goto live_menu
echo ${cls}
goto ${flavor} ||

:pop-18.04-default-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/58-600b3df6/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/58-600b3df6/
goto 18.04-boot
:pop-18.04-nvidia-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/58-316fd7f7/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/58-316fd7f7/
goto 18.04-boot
:pop-19.10-default-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/10-bdc3fe0a/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/10-bdc3fe0a/
goto 19.10-boot
:pop-19.10-nvidia-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/10-3882d464/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/10-3882d464/
goto 19.10-boot
:pop-20.04-default-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/5-66b7e861/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/5-66b7e861/
goto 20.04-boot
:pop-20.04-nvidia-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/5-6efd1d75/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/5-6efd1d75/
goto 20.04-boot
:pop-21.04-default-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/21.04-62feddb8/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/21.04-62feddb8/
goto 21.04-boot
:pop-21.04-nvidia-squash
set squash_url ${live_endpoint}/ubuntu-squash/releases/download/21.04-601c9b89/filesystem.squashfs
set kernel_url ${live_endpoint}/ubuntu-squash/releases/download/21.04-601c9b89/
goto 21.04-boot

:18.04-boot
imgfree
kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=http fetch=${squash_url} initrd=initrd.magic ${cmdline}
initrd ${kernel_url}initrd
boot

:20.04-boot
:21.04-boot
imgfree
kernel ${kernel_url}vmlinuz ip=dhcp boot=casper netboot=url url=${squash_url} initrd=initrd.magic ${cmdline}
initrd ${kernel_url}initrd
boot

:live_exit
clear menu
exit 0
```

**Reply by Yethal** [reply]
@NiKiZe did you get a chance to look at this? It's the same for 21.10

**Reply by NiKiZe** [reply]
Well as far as I can see the OS (kernel) boots just fine, it then fails to mount over NFS.
Eighter take this with netboot.xyz or with pop. Not really an iPXE issue?

**Reply by Yethal** [reply]
I did take this with netboot.xyz, they redirected me here. Perhaps reporting this directly to Pop will yield better result

**Reply by NiKiZe** [reply]
This is an issue with the scripts/configuration. Is there an original netboot.xyz issue that you can link to?

**Reply by Yethal** [reply]
It seems that I misremembered things, I went through my github history and found the original issue but it was filed in this repo and not at netboots https://github.com/ipxe/ipxe/issues/532. However I went through netboot's repo and found this issue reported by someone else here: https://github.com/netbootxyz/netboot.xyz/issues/1067
